packages/editor/src/services/SceneThumbnailState.tsx

```diff
@@ -0,0 +1,126 @@
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Infinite Reality Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Infinite Reality Engine team.
+
+All portions of the code written by the Infinite Reality Engine team are Copyright © 2021-2023 
+Infinite Reality Engine. All Rights Reserved.
+*/
+
+import {
+  blurAndScaleImageData,
+  convertImageDataToKTX2Blob,
+  imageDataToBlob
+} from '@ir-engine/engine/src/scene/classes/ImageUtils'
+import { SceneSettingsComponent } from '@ir-engine/engine/src/scene/components/SceneSettingsComponent'
+import { defineState, getMutableState, getState } from '@ir-engine/hyperflux'
+import { commitProperty } from '../components/properties/Util'
+import { uploadProjectFiles } from '../functions/assetFunctions'
+import { takeScreenshot } from '../functions/takeScreenshot'
+import { generateEnvmapBake } from '../functions/uploadEnvMapBake'
+import { EditorState } from './EditorServices'
+
+export const SceneThumbnailState = defineState({
+  name: 'ee.editor.SceneThumbnailState',
+  initial: () => ({
+    thumbnailURL: null as string | null,
+    thumbnail: null as File | null,
+    uploadingThumbnail: false,
+    loadingScreenURL: null as string | null,
+    loadingScreenImageData: null as ImageData | null,
+    uploadingLoadingScreen: false,
+    resolution: 2048
+  }),
+  createThumbnail: async () => {
+    const thumbnailBlob = await takeScreenshot(512, 320, 'jpeg')
+    if (!thumbnailBlob) return
+    const thumbnailURL = URL.createObjectURL(thumbnailBlob)
+    const sceneName = getState(EditorState).sceneName!.split('.').slice(0, -1).join('.')
+    const file = new File([thumbnailBlob!], sceneName + '.thumbnail.jpg')
+    getMutableState(SceneThumbnailState).merge({
+      thumbnailURL,
+      thumbnail: file
+    })
+  },
+  uploadThumbnail: async () => {
+    const sceneThumbnailState = getMutableState(SceneThumbnailState)
+    if (!sceneThumbnailState.thumbnail.value) return
+    sceneThumbnailState.uploadingThumbnail.set(true)
+    const editorState = getState(EditorState)
+    const projectName = editorState.projectName!
+    const currentSceneDirectory = getState(EditorState).scenePath!.split('/').slice(0, -1).join('/')
+    const { promises } = uploadProjectFiles(projectName, [sceneThumbnailState.thumbnail.value], [currentSceneDirectory])
+    const [[savedThumbnailURL]] = await Promise.all(promises)
+    commitProperty(SceneSettingsComponent, 'thumbnailURL')(savedThumbnailURL)
```

Comment:
we need to pass the correct entity into this explicitly, otherwise if the entity is not selected, this function crashes

Reply:
done